qiskit/compiler/transpiler.py

```diff
@@ -325,8 +325,14 @@ def callback_func(**kwargs):
             backend_properties = target_to_backend_properties(target)
     # If target is not specified and any hardware constraint object is
     # manually specified then do not use the target from the backend as
-    # it is invalidated by a custom basis gate list or a custom coupling map
-    elif basis_gates is not None or coupling_map is not None:
+    # it is invalidated by a custom basis gate list, custom coupling map,
+    # custom dt or custom instruction_durations
+    elif (
+        basis_gates is not None
+        or coupling_map is not None
+        or dt is not None
+        or instruction_durations is not None
```

Comment:
I'm wondering if we should expand this to include `backend_properties` , `inst_map`, or `timing_constraints` are set too? Those fields also have overlap with the target.

Reply:
We probably should expand it to `backend_properties`, but `inst_map` and `timing_constraints` are actually still respected for `BackendV2` inputs. `inst_map` is applied [here](https://github.com/ElePT/qiskit/blob/a28394cb78f390c9874aa16134c406ac30afac7e/qiskit/compiler/transpiler.py#L351), but I still don't know how is `timing_constraints` applied. In fact,`timing_constraints` also takes precedence over `target`, which should not be the case.

I have started making these tables to try to track the priority of each constraint in each setting. The starting point (before this PR) is:

| vs               |   target              |   backendV1              |   backendV2           |
|--------------------------|-----------------------|--------------------------|-----------------------|
|   basis_gates            |   target              |   basis_gates            |   basis_gates         |
|   coupling_map           |   target              |   coupling_map           |   coupling_map        |
|   instruction_durations  |   target              |   instruction_durations  |   **backendV2**           |
|   inst_map               |   target              |   inst_map               |   inst_map            |
|   dt                     |   target              |   dt                     |   **backendV2**           |
|   timing_constraints     |   **timing_constraints**  |   timing_constraints     |   timing_constraints  |
|   backend_properties     |   target              |   backend_properties     |   **backendV2**           |

Each cell shows the "winner" between the corresponding row and column header if both are provided (I have a test file where I see who wins in each case). The expected behavior is that the target should always win, and the backends should always loose. This PR I think should fix all of the "backendV2" anomalies (marked in bold font). So I think after this PR the table should look like this:

| vs              |   target              |   backendV1              |   backendV2           |
|--------------------------|-----------------------|--------------------------|-----------------------|
|   basis_gates            |   target              |   basis_gates            |   basis_gates         |
|   coupling_map           |   target              |   coupling_map           |   coupling_map        |
|   instruction_durations  |   target              |   instruction_durations  |   instruction_durations           |
|   inst_map               |   target              |   inst_map               |   inst_map            |
|   dt                     |   target              |   dt                     |   dt           |
|   timing_constraints     |   **timing_constraints**  |   timing_constraints     |   timing_constraints  |
|   backend_properties     |   target              |   backend_properties     |   backend_properties       |

I'll try to see if I can also fix the `timing_constraints` anomaly.